cmds/modules/noded/main.go

```diff
@@ -191,6 +197,28 @@ func action(cli *cli.Context) error {
 		return version, nil
 	})
 
+	bus.WithHandler("zos.system.diagnostics", func(ctx context.Context, payload []byte) (interface{}, error) {
+		results := Diagnostics{
+			SystemStatusOk: true,
+			Modules:        make(map[string]zbus.Status),
+		}
+
+		for module := range zbusdebug.PossibleModules {
+			status, err := redis.Status(ctx, module)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to get status of module %s", module)
+			}
+			for _, worker := range status.Workers {
+				if worker.State != "free" {
+					results.SystemStatusOk = false
```

Comment:
I'm not sure if the comparison against WorkerFree by itself is enough, i think the check should be in status, err := ... and then do if err -> systemOk is false instead (you shouldn't return there)